packages/origin-247-certificate/src/evented-certificate/certificate.facade.ts

```diff
@@ -0,0 +1,167 @@
+import { Injectable, Inject } from '@nestjs/common';
+import {
+    CERTIFICATE_COMMAND_REPOSITORY,
+    CertificateCommandRepository
+} from './repositories/CertificateCommand/CertificateCommand.repository';
+import {
+    CERTIFICATE_EVENT_REPOSITORY,
+    CertificateEventRepository
+} from './repositories/CertificateEvent/CertificateEvent.repository';
+import { CertificateService } from '../certificate.service';
+import { QueryBus, EventBus } from '@nestjs/cqrs';
+import { GetAllCertificatesQuery, GetCertificateQuery } from '@energyweb/issuer-api';
+import { CertificateEventType } from './events/Certificate.events';
+
+import {
+    ICertificate,
+    IClaimCommand,
+    IIssueCommand,
+    ITransferCommand,
+    ISuccessResponse,
+    BlockchainActionType,
+    BlockchainAction,
+    IIssuedCertificate,
+    IIssueCommandParams
+} from '../types';
+import {
+    CertificateIssuedEvent,
+    CertificateTransferredEvent,
+    CertificateClaimedEvent
+} from './events/Certificate.events';
+
+const CertificateEventVersion = 1;
+
+type CertificateCommand = IIssueCommand<unknown> | ITransferCommand | IClaimCommand;
+@Injectable()
+export class CertificateFacade<T = null> {
```

Comment:
I think this should reflect the module's name. And on that: I'm not if "evented-certificate" is the right name

Reply:
Renamed to OffchainCertificateModule and OffchainCertificateFacade